sample-android-app/src/main/java/com/badoo/reaktive/sample/android/MainActivity.kt

```diff
@@ -1,37 +1,62 @@
 package com.badoo.reaktive.sample.android
 
+import android.annotation.SuppressLint
 import android.os.Bundle
 import android.support.v7.app.AppCompatActivity
 import android.widget.Button
 import android.widget.TextView
+import com.badoo.reaktive.disposable.CompositeDisposable
+import com.badoo.reaktive.observable.map
 import com.badoo.reaktive.observable.observableFromFunction
 import com.badoo.reaktive.observable.observeOn
+import com.badoo.reaktive.observable.onErrorReturn
 import com.badoo.reaktive.observable.startWithValue
 import com.badoo.reaktive.observable.subscribe
 import com.badoo.reaktive.observable.subscribeOn
-import com.badoo.reaktive.scheduler.ioScheduler
+import com.badoo.reaktive.observable.switchMap
+import com.badoo.reaktive.scheduler.computationScheduler
 import com.badoo.reaktive.scheduler.mainScheduler
+import com.badoo.reaktive.subject.publish.publishSubject
 import java.text.SimpleDateFormat
-import java.util.Date
+import java.util.*
 
 class MainActivity : AppCompatActivity() {
 
+    private val clickSubject = publishSubject<Unit>()
+    private val disposables = CompositeDisposable()
+
+    @SuppressLint("SetTextI18n")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         setContentView(R.layout.activity_main)
 
         val textView = findViewById<TextView>(R.id.text)
 
-        findViewById<Button>(R.id.button).setOnClickListener {
-            observableFromFunction<String> {
-                Thread.sleep(1000L)
-                SimpleDateFormat.getDateTimeInstance().format(Date())
+        findViewById<Button>(R.id.button).setOnClickListener { clickSubject.onNext(Unit) }
+
+        disposables += clickSubject
+            .switchMap {
+                observableFromFunction {
+                    Thread.sleep(1000L)
+                    return@observableFromFunction Date()
+                }
+                    .map(SimpleDateFormat.getDateTimeInstance()::format)
+                    .subscribeOn(computationScheduler)
+                    .onErrorReturn { throwable -> throwable.toString() }
+                    .startWithValue("Loading...")
             }
-                .subscribeOn(ioScheduler)
-                .observeOn(mainScheduler)
-                .startWithValue("Loading...")
-                .subscribe(onNext = textView::setText)
-        }
+            .observeOn(mainScheduler)
+            .subscribe(
+                onNext = textView::setText,
+                onError = { throwable -> textView.text = throwable.toString() },
```

Comment:
You have `onErrorReturn` operator above, what is the point of handling errors then?

Reply:
I modified `subscribe` a little bit. You are right that we didn't expect onError/onComplete calls. 
So I think it will be better to throw IllegalStateException when onError/onComplete is called. Fatal crash tells us that Observable (bussiness logic) wasn't properly composed.

Reply:
But it is impossible to receive `onError` call, since you have `onErrorReturn`. Only if there are some bugs in Reaktive lib. Same for `onComplete`, it will not be called by definition. Better to just remove the callbacks to avoid confusion.